models/enterprise_sonic/snmp/sonic_snmp.yml

```
@@ -1,53 +1,280 @@
+                elements: dict
+                suboptions:
+                  security_model:
+                    description:
+                      - Security model used for user authentication
+                    type: str
+                    required: true
+                    choices:
+                      - any
+                      - v2c
+                      - v3
+                  security_level:
+                    description:
+                      - Minimum security level under which the access rights apply. Only used if security model is v3
+                    type: str
+                    choices:
+                      - no-auth-no-priv
+                      - auth-no-priv
+                      - auth-priv
+                  read_view:
+                    description:
+                      - Name of the read view
+                    type: str
+                  write_view:
+                    description:
+                      - Name of the write view
+                    type: str
+                  notify_view:
+                    description:
+                      - Name of the notify view
+                    type: str  
+          host: 
+            description:
+              - List of SNMP host configuration
+            type: list
+            elements: dict
+            suboptions:
+              ip:
+                description:
+                  - IPv4 or IPv6 address
+                type: str
+                required: true
+              community:
+                description:
+                  - SNMP community for the host
+                type: str
+              user:
+                description:
+                  - SNMP user
+                type: dict
+                suboptions:
+                  name:
+                    description:
+                    - Name of the user
+                    type: str
+                    required: true
+                  security_level:
+                    description:
+                      - Security level of the user
+                    type: str
+                    choices:
+                      - auth
+                      - noauth
+                      - priv
+              port:
+                description:
+                  - UDP destination port, range 1024-65535
+                type: int
+              source_interface:
+                description:
+                  - Name of source interface
+                type: str
+              tag:
+                description: 
+                  - Notification type
+                  - If "inform" is selected, a "retries" value/option between 1 and 255 can be specified
+                type: str
+                choices:
+                  - inform
+                  - trap
+              retries:
+                description:
+                  - Number of retries that is only used when "inform" choice is speccified.
+                  - Has a range between 1 and 255
```

Please add the type for the "retries" option:

```suggestion
                  - Has a range between 1 and 255
                type: int
```